modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -0,0 +1,260 @@
+        return toDataType(ResolvableType.forField(field));
+    }
+
+    public static Field getField(Class<?> clazz, Parameter parameter)
+    {
+        return getField(clazz, getMemberName(parameter, parameter.getName()), parameter.getType().getRawType());
+    }
+
+    public static Field getField(Class<?> clazz, String name, Class<?> type)
+    {
+        Collection<Field> candidates = getAllFields(clazz, withName(name), withTypeAssignableTo(type));
+        return CollectionUtils.isEmpty(candidates) ? null : candidates.iterator().next();
+    }
+
+    public static boolean hasDefaultConstructor(Class<?> clazz)
+    {
+        return ClassUtils.getConstructor(clazz, new Class[] {}) != null;
+    }
+
+    private static DataType toDataType(ResolvableType type)
+    {
+        Class<?> rawClass = type.getRawClass();
+        ResolvableType[] generics = type.getGenerics();
+
+        if (isOperation(rawClass))
+        {
+            return DataType.of(Operation.class);
+        }
+
+        if (List.class.isAssignableFrom(rawClass))
+        {
+            if (!ArrayUtils.isEmpty(generics) && isOperation(generics[0].getRawClass()))
+            {
+                return DataType.of(rawClass, Operation.class);
+            }
+        }
+
+        return DataType.of(rawClass, toRawTypes(generics));
+    }
+
+    private static boolean isOperation(Class<?> rawClass)
+    {
+        return NestedProcessor.class.isAssignableFrom(rawClass);
+    }
+
+    private static Class<?>[] toRawTypes(ResolvableType[] resolvableTypes)
+    {
+        Class<?>[] types = new Class<?>[resolvableTypes.length];
+        for (int i = 0; i < resolvableTypes.length; i++)
+        {
+            types[i] = resolvableTypes[i].getRawClass();
+        }
+
+        return types;
+    }
+
+    public static void checkInstantiable(Class<?> declaringClass)
+    {
+        checkInstantiable(declaringClass, true);
+    }
+
+    public static void checkInstantiable(Class<?> declaringClass, boolean requireDefaultConstructor)
+    {
+        checkArgument(declaringClass != null, "declaringClass cannot be null");
+
+        if (requireDefaultConstructor)
+        {
+            checkArgument(hasDefaultConstructor(declaringClass), String.format("Class %s cannot be instantiated since it doesn't have a default constructor", declaringClass.getName()));
+        }
+
+        checkArgument(!declaringClass.isInterface(), String.format("Class %s cannot be instantiated since it's an interface", declaringClass.getName()));
+        checkArgument(!Modifier.isAbstract(declaringClass.getModifiers()), String.format("Class %s cannot be instantiated since it's abstract", declaringClass.getName()));
+    }
+
+    public static boolean isIgnored(AccessibleObject object)
+    {
+        return object == null || object.getAnnotation(Ignore.class) != null;
+    }
+
+    public static boolean isRequired(AccessibleObject object)
+    {
+        return object.getAnnotation(Optional.class) == null;
```

Comment:
Methods in this class have check for valid arguments. Would be good if all the public methods behave in the same way